### ISSUE TITLE
Payments > Deposits doesn't make sense with a negative account balance

### ISSUE THREAD

**Issue by aheckler** [Automattic/woocommerce-payments#3302]
There's no one specific bug here, but many? TLDR: the Payments > Deposits page needs looked at to make sure that it makes sense _even when the merchant's account balance goes negative_. 

For example, in 4429343-zen, the merchant has a negative balance of ~$500 USD. Here's what their Deposits page looks like in our plugin:

<img width="962" alt="Markup 2021-11-05 at 13 20 09" src="https://user-images.githubusercontent.com/5512652/140556881-dc60b8f3-af2b-45c0-a91d-8ea1c683da4f.png">

Several issues spring to mind here:

- It's confusing to call these "Estimated", since as far as I can tell, they are exact figures we're pulling right from Stripe.
- It's confusing that the "Estimated" dates are in the past.
- Most importantly, because they have a negative account balance, Stripe won't actually be crediting anything to their debit card, just crediting their Stripe account balance with these amounts. When they make enough new sales to get back to a positive account balance, that's when Stripe will start paying out again. There's no indication of that reality on this screen though. It more or less acts like these credits/debits are about to happen, when they won't be.

**Comment by htdat** [comment]
@LevinMedia - would you mind having a look from the design perspective?

**Comment by LevinMedia** [comment]
@htdat "Estimated" should **_only_** be used prior to the deposit being transfered, or withheld from the merchant. Once the deposit (or withdrawl) is successful the status should say "Paid" if it went through successfully. 

If I'm understanding the second issue correctly, it sounds like it may be helpful to have different language (other than "withdrawal" and "Paid") for the status of a withdrawal made against a debit card?

**Comment by htdat** [comment]
> "Estimated" should only be used prior to the deposit being transfered, or withheld from the merchant. Once the deposit (or withdrawl) is successful the status should say "Paid" if it went through successfully.

Yes, that's correct! And we use "estimated" because we need to calculate the info from Stripe as we use `Delay payouts by x days`. 

> If I'm understanding the second issue correctly, it sounds like it may be helpful to have different language (other than "withdrawal" and "Paid") for the status of a withdrawal made against a debit card?

It may be a bit tricky as Stripe can debit bank accounts but can not debit cards. But yeah, I agree that we may have a better word here.

From looking at this account transactions history, here is what I understood: 

- This account did refund for two orders on Oct 21.
- This account got deposit as planned on Oct 22. Note this account has `Delay payouts by 2 days` setting.
- Stripe could not debit the merchant's Visa card.
- Next successful payments on the following days (Oct 25, Nov 3, and Nov 5) can not offset the refund amount on Oct 22. Stripe did not really send deposits because of that. 
- Only until Nov 10, the deposit became positive after deducting the refund amount. 

@dechov - you might have some insights here for this issue as I've just seen you and @diegocurbelo discussed in p1639538471143600-slack-CGGCLBN58

**Comment by LevinMedia** [comment]
@htdat - having thought more about this: in the case of the debit cards, rather than "Withdrawal" I think we could use the words "Balance debit" when there's "withdrawal" against a debit card.

Deposit should still work as a term regardless if the balance is positive or negative

**Comment by darcie** [comment]
Feedback from 5230740-zen. They had an advance but are still showing an estimated deposit. The information in Payments > Deposits doesn't match what transactions were counted by Stripe as paid out to them. It looks like they have an upcoming deposit that was missed, when they've been fully paid out for that by the instant payment.

**Comment by csmcneill** [comment]
5528040-zen

**Comment by anu-rock** [comment]
5632282-zd-woothemes

**Comment by haszari** [comment]
We should consider this in upcoming project refreshing deposits UI.

**Comment by james-allan** [comment]
I ran into this issue just recently and I think I discovered the cause is on Server. 
When the merchant's deposit schedule is set to daily, server groups transactions into daily estimated deposits using the balance transaction's `available_on` property. 

Sometimes the refund balance transaction object has an `available_on` date in a `2023-04-13 07:02:33` or `2023-04-21 00:00:00` format. Note the 00:00:00 vs 07:02:00 precision difference.

I don't know what causes this differentiation on Stripe's end. It could be any one of these potential factors:

- the transaction I refunded was already in the available balance and so the refund was also immediately “available”. Whereas the Apr 21 one I observed is available on that date because that is  when the corresponding original transaction was becoming available. ie the refund follows the original transaction’s availability.
- depending on the balance. The refund is available immediately if the available balance can cover it. Otherwise it has a future available_on date.

When a refund has a date that includes hours:minutes:seconds precision (eg `07:02:33`) it gets grouped into its own deposit whereas other transactions on that day will be grouped to the day (`00:00:00`).

The relevant code can be found in the `get_closest_payout_date()` function on server under the daily deposit schedule condition.

**Comment by brucealdridge** [comment]
This issue is specifically about negative estimated deposits in the past (caused by refunds and disputes). I've gone ahead and retitled to make the underlying problem clearer at a glance

This work is planned for the next sprint.

**Comment by Jinksi** [comment]
Resolved by #7604 since we no longer display `estimated` deposits
